frontend/src/component/releases/ReleasePlanTemplate/EditReleasePlanTemplate.tsx

```diff
@@ -0,0 +1,139 @@
+import { useUiFlag } from 'hooks/useUiFlag';
+import { usePageTitle } from 'hooks/usePageTitle';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import { useReleasePlanTemplateInstance } from 'hooks/api/getters/useReleasePlanTemplates/useReleasePlanTemplateInstance';
+import FormTemplate from 'component/common/FormTemplate/FormTemplate';
+import { useTemplateForm } from '../hooks/useTemplateForm';
+import { TemplateForm } from './TemplateForm';
+import { Box, Button, Card, styled } from '@mui/material';
+import { UpdateButton } from 'component/common/UpdateButton/UpdateButton';
+import { ADMIN } from '@server/types/permissions';
+import { useNavigate } from 'react-router-dom';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import useToast from 'hooks/useToast';
+import useReleasePlanTemplatesApi from 'hooks/api/actions/useReleasePlanTemplatesApi/useReleasePlanTemplatesApi';
+
+const StyledForm = styled('form')(() => ({
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+}));
+
+const StyledMilestoneCard = styled(Card)(({ theme }) => ({
+    marginTop: theme.spacing(2),
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    boxShadow: 'none',
+    border: `1px solid ${theme.palette.divider}`,
+    [theme.breakpoints.down('sm')]: {
+        justifyContent: 'center',
+    },
+    transition: 'background-color 0.2s ease-in-out',
+    backgroundColor: theme.palette.background.default,
+    '&:hover': {
+        backgroundColor: theme.palette.neutral.light,
+    },
+    borderRadius: theme.shape.borderRadiusMedium,
+}));
+
+const StyledMilestoneCardBody = styled(Box)(({ theme }) => ({
+    padding: theme.spacing(3, 2, 3, 2),
+}));
+
+const StyledMilestoneCardTitle = styled('span')(({ theme }) => ({
+    fontWeight: theme.fontWeight.bold,
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+const StyledButtonContainer = styled('div')(() => ({
+    marginTop: 'auto',
+    display: 'flex',
+    justifyContent: 'flex-end',
+}));
+
+const StyledCancelButton = styled(Button)(({ theme }) => ({
+    marginLeft: theme.spacing(3),
+}));
+
+export const EditReleasePlanTemplate = () => {
+    const releasePlansEnabled = useUiFlag('releasePlans');
+    if (!releasePlansEnabled) {
+        return null;
+    }
```

Comment:
My gut feeling is that this will cause you issues. Usually you should do this right before the component's `return` at the end, so the same number of hooks always run for each render.

Reply:
Oh, interesting! thanks!